### PR TITLE
Bug Fix #11240. Now Array, Seq, ArraySeq sortWith return the sorted a…

### PR DESCRIPTION
**Pull request by ashwinbhaskar**
…rray instead of the old array.

scala/bug#11240

### CODE REVIEW

File: src/library/scala/collection/ArrayOps.scala [hunk 1]
```diff
@@ -491,8 +491,15 @@ final class ArrayOps[A](val xs: Array[A]) extends AnyVal {
       case xs: Array[AnyRef] =>
         val a = Arrays.copyOf(xs, len); Arrays.sort(a, ord.asInstanceOf[Ordering[AnyRef]]); a
       case xs: Array[Int] =>
-        if(ord eq Ordering.Int) { val a = Arrays.copyOf(xs, len); Arrays.sort(a); a }
-        else boxed
+        println("in array pattern matching. Matched as Array ofInt")
```
Comment (He-Pin) [inline src/library/scala/collection/ArrayOps.scala]: add a test instead of println:)

Comment (ashwinbhaskar) [reply src/library/scala/collection/ArrayOps.scala]: Sorry about that. Should I remove it and send a pull request again?

Comment (NthPortal) [reply src/library/scala/collection/ArrayOps.scala]: nah, you can just force-push the changes

Comment (He-Pin) [reply src/library/scala/collection/ArrayOps.scala]: Commit,rebase,pick and fix up,push -f

Comment (ashwinbhaskar) [reply src/library/scala/collection/ArrayOps.scala]: @NthPortal @hepin1989 done

Comment (ashwinbhaskar) [reply src/library/scala/collection/ArrayOps.scala]: @hepin1989 Who would be merging this pull request?

Comment (He-Pin) [reply src/library/scala/collection/ArrayOps.scala]: @ashwinbhaskar Teamember

**Review by NthPortal — APPROVED** [review]
LGTM

File: src/library/scala/collection/ArrayOps.scala (hunk 1 above)
Comment (ashwinbhaskar) [reply src/library/scala/collection/ArrayOps.scala]: @hepin1989 @NthPortal @adriaanm Can I know how many LGTMs before merge happens? Just curious and trying to understand how things go about here:) This is my first contribution to scala core lib

Comment (SethTisue) [reply src/library/scala/collection/ArrayOps.scala]: welcome!

> how many LGTMs before merge happens

it really depends on how large and/or how controversial the change is. typically we want at least one green review and then a different person will do the final merge, but some PRs are easier than that and some are harder.

**Comment by SethTisue** [comment]
the code change looks good, but normally we also want to have test coverage for a bug fix. is that doable here?

**Comment by ashwinbhaskar** [comment]
@SethTisue I have added test coverage now

**Comment by ashwinbhaskar** [comment]
@SethTisue is it good now?

**Comment by adriaanm** [comment]
Thanks, good catch!

**Pull request merged**